src/codec/rtu.rs

```diff
@@ -300,36 +296,28 @@ impl Decoder for ServerCodec {
     type Error = Error;
 
     fn decode(&mut self, buf: &mut BytesMut) -> Result<Option<RequestAdu>> {
-        self.decoder
-            .decode(buf)
-            .and_then(|frame| {
-                if let Some((slave_id, pdu_data)) = frame {
-                    let hdr = Header { slave_id };
-                    // Decoding of the PDU should are unlikely to fail due
-                    // to transmission errors, because the frame's bytes
-                    // have already been verified with the CRC.
-                    RequestPdu::try_from(pdu_data)
-                        .map(|pdu| {
-                            Some(RequestAdu {
-                                hdr,
-                                pdu,
-                                disconnect: false,
-                            })
-                        })
-                        .map_err(|err| {
-                            // Unrecoverable error
-                            log::error!("Failed to decode request PDU: {}", err);
-                            err
-                        })
-                } else {
-                    Ok(None)
-                }
-            })
-            .map_err(|_| {
-                // Decoding the transport frame is non-destructive and must
-                // never fail!
-                unreachable!();
-            })
+        if let Some((slave_id, pdu_data)) = self.decoder.decode(buf)? {
```

Comment:
A `let-else` binding with an early return would improve readability.

Reply:
Changed. Like above, I refrained from introducing that earlier, because it's only supported since Rust 1.65.
Even worse, while it's supported, it's currently not picked up by `rustfmt`, so people will introduce their own coding styles around let-else. I've written it in a way that I consider meaningful, but a future `rustfmt` may reformat that.

Along with this change, I've added a line `rust-version = "1.65"` to `Cargo.toml` and added an entry to the changelog.